src/main.rs

```diff
@@ -36,11 +51,29 @@ struct TypeSizeCmd {
 
 impl TypeSizeCmd {
     fn new(old_args: impl Iterator<Item = String>) -> Self {
-        let cargo_subcommand = "check";
+        let mut cargo_subcommand = "check";
         let args = vec![];
         let mut typesize_args: Vec<String> = vec![];
 
-        typesize_args.extend(old_args);
+        for arg in old_args {
+            match arg.as_str() {
+                "--build" => {
+                    cargo_subcommand = "build";
+                    continue;
+                }
+                "--check" => {
+                    cargo_subcommand = "check";
+                    continue;
+                }
+                "--test" => {
+                    cargo_subcommand = "test";
+                    continue;
+                }
+                "--" => break,
+                _ => {}
+            }
+            typesize_args.push(arg);
```

Comment:
You need to be pushing to the `args` vec for these arguments to be seen by `cargo`
```suggestion
            args.push(arg);
```
`typesize_args` is converted into an environment variable  to be passed to `driver.rs`

`args`  are passed to cargo as CLI args.

For example `cargo typesize build --example pprint`, the `["--example", "pprint"]` parts need to go into the `args` vec so that we can pass that to cargo